src/osd/PrimaryLogPG.cc

```diff
@@ -4698,9 +4702,19 @@ struct C_ExtentCmpRead : public Context {
 					     &read_bl, maybe_crc, size,
 					     osd, soid, flags)) {
   }
+  ~C_ExtentCmpRead() override {
+    delete fill_extent_ctx;
+  }
 
   void finish(int r) override {
-    fill_extent_ctx->complete(r);
+    if (r == -ENOENT) {
+      osd_op.rval = 0;
+      read_bl.clear();
+      delete fill_extent_ctx;
```

Comment:
Yes -- but that was just to silence coverity. If I avoid the deletion here, I would need to move the nullptr assignment within the else branch. I can do it -- but at least it's consistent that the object is destructed within this method now.